### PR TITLE
Core/EventMap: Finish std::chrono-fication

### PR DESCRIPTION
**Pull request by Carbenium**
**Changes proposed:**

-  EventMap: finish std::chrono-fication
-  EventMap: Unify semantics of DelayEvents
`DelayEvents(Milliseconds delay)` had different semantics than `DelayEvents(Milliseconds delay, uint32 group)`.
The first method delayed the events only in the case the internal timer already ticked at least for the amount of delay. In contrast the latter method delayed events regardless of the internal timer value.
Use the latter semantics for `DelayEvents(Milliseconds delay)` as well which makes the outcome more predictable.
Callers were examined. No usage was found which depends on this edge case.

-  EventMap: Add some more tests

**Target branch(es):** 3.3.5/master

- [x] 3.3.5
- [ ] master

**Tests performed:** build + unit tests

### CODE REVIEW

File: src/common/Utilities/EventMap.cpp [hunk 1]
```diff
@@ -94,7 +94,18 @@ uint32 EventMap::ExecuteEvent()
...
+
+    EventStore delayed;
+
+    for (EventStore::iterator itr = _eventMap.begin(); itr != _eventMap.end();)
+    {
+        delayed.insert(EventStore::value_type(itr->first + delay, itr->second));
+        _eventMap.erase(itr++);
+    }
+
+    _eventMap.insert(delayed.begin(), delayed.end());
```
Comment (Shauren) [inline src/common/Utilities/EventMap.cpp:108]: Maybe something like this?

```suggestion
    EventStore delayed = std::move(_eventMap);

    for (EventStore::iterator itr = delayed.begin(); itr != delayed.end();)
    {
        EventStore::node_type node = delayed.extract(itr++);
        node.key() = node.key() + delay;
        _eventMap.insert(_eventMap.end(), std::move(node));
    }
```

Comment (Carbenium) [reply src/common/Utilities/EventMap.cpp:108]: Hah, `extract`. That's a new one. Thanks for the suggestion!
